src/SDK/Client/ClientApps.ts

```diff
@@ -3,71 +3,116 @@ import Identifier from "@dashevo/dpp/lib/Identifier";
 /**
  * Interface for ClientApps
  */
-export interface ClientAppsOptions {
-    [name: string]: ClientAppDefinitionOptions,
-}
 
-interface ClientAppDefinitionOptions {
-    contractId: Identifier|string,
+export interface ClientAppDefinitionOptions {
+    contractId: Identifier | string,
     contract?: any
-}
+    aliases?: [string],
+    alias?: string
+};
 
-interface ClientAppDefinition {
+export interface ClientAppDefinition {
     contractId: Identifier,
-    contract?: any
+    contract: any,
+    aliases: [string]
 }
 
-type ClientAppsList = Record<string, ClientAppDefinition>;
+type ClientAppsList = Record<Identifier, ClientAppDefinition>;
 
 export class ClientApps {
     private apps: ClientAppsList = {};
 
-    constructor(apps: ClientAppsOptions = {}) {
-        Object.entries(apps).forEach(([name, definition]) => this.set(name, definition));
+    constructor(apps: Array<ClientAppDefinitionOptions> = []) {
+        apps && apps.forEach((appDefinition)=> this.set(appDefinition.contractId, appDefinition));
     }
 
+    getApps(){
+        return this.apps;
+    }
     /**
      * Set app
      *
-     * @param {string} name
-     * @param {ClientAppDefinitionOptions} definition
+     * @param {string|Identifier} contractId
+     * @param {ClientAppDefinitionOptions} appProperties
      */
-    set(name: string, definition: ClientAppDefinitionOptions) {
-        definition.contractId = Identifier.from(definition.contractId);
+    set(contractId: string | Identifier, appProperties) {
+        const identifier = Identifier.from(contractId);
 
-        this.apps[name] = definition;
+        const definition = this.apps[identifier] || {
+            contractId: identifier.toString(),
+            contract: null,
+            aliases: []};
+
+        if (appProperties.contract) definition.contract = appProperties.contract;
+        if (appProperties.alias) {
+            definition.aliases.push(appProperties.alias);
+        }
+        if (appProperties.aliases) definition.aliases.push(...appProperties.aliases);
+        this.apps[Identifier.from(identifier)] = definition;
     }
 
     /**
-     * Get app definition by name
+     * Get app definition by name or identifier
      *
-     * @param {string} name
+     * @param {string|Identifier} query
      * @return {ClientAppDefinition}
      */
-    get(name: string): ClientAppDefinition {
-        if (!this.has(name)) {
-            throw new Error(`Application with name ${name} is not defined`);
+    get(query: string | Identifier) {
+        let identifier;
+        try {
+            identifier = Identifier.from(query);
+        } catch (e) {
+            const appSearch = Object.entries(this.apps).find((el) => {
```

Comment:
We can't catch all errors here. We need to check for a specific one and throw in other cases.

Reply:
I don't get this one, Identifier.from() only throw IdentifierError where encoding is not the good one. 
Which is then when we provide another way to parse it.
What other error is happening ?

Reply:
It could be a syntax error or anything else. I just suggest checking `IdentifierError` and re-throw others.

Reply:
This has been updated 👍